File: atlaskv-java-sdk/src/main/java/com/atlaskv/sdk/models/Lease.java
package com.atlaskv.sdk.models;

import java.util.Set;

/**
 * Immutable representation of an active distributed lease.
 *
 * @param leaseId      unique lease ID
 * @param durationMs   duration of the lease in milliseconds
 * @param expiryTimeMs absolute expiry timestamp in milliseconds
 * @param keys         set of keys currently associated with this lease
 */
public record Lease(
        String leaseId,
        long durationMs,
        long expiryTimeMs,
        Set<String> keys
) {}

File: atlaskv-ts-sdk/src/models/Lease.ts
/**
 * Represents an active distributed lease in AtlasKV.
 */
export interface Lease {
  /** The unique identifier of the lease. */
  leaseId: string;
  /** The configured lease duration in milliseconds. */
  durationMs: number;
  /** The absolute epoch timestamp in milliseconds when the lease expires. */
  expiryTimeMs: number;
  /** The list of keys associated with and governed by this lease. */
  keys: string[];
}
